ledger/participant-integration-api/src/main/scala/platform/apiserver/RateLimitingInterceptor.scala

```diff
@@ -13,11 +13,26 @@ import io.grpc._
 import io.grpc.protobuf.StatusProto
 import org.slf4j.LoggerFactory
 
-private[apiserver] final class RateLimitingInterceptor(metrics: Metrics, config: RateLimitingConfig)
-    extends ServerInterceptor {
+import java.lang.management.{MemoryMXBean, MemoryPoolMXBean, MemoryType}
+
+private[apiserver] final class RateLimitingInterceptor(
+    metrics: Metrics,
+    config: RateLimitingConfig,
+    memoryPoolMxBeans: List[MemoryPoolMXBean],
+    memoryMxBean: MemoryMXBean,
+) extends ServerInterceptor {
 
   private val logger = LoggerFactory.getLogger(getClass)
 
+  private val tenuredMemoryPool: Option[MemoryPoolMXBean] =
+    memoryPoolMxBeans.find(p =>
+      p.getType == MemoryType.HEAP && p.isCollectionUsageThresholdSupported
+    )
+
+  tenuredMemoryPool.foreach { p =>
```

Comment:
To increase robustness of the solution I would suggest dealing explicitly with 2 possible future error cases:
* we encounter a GC that doesn't have any tenured pool
* we encounter a GC with more than one tenured pool

In both cases the mechanism should be disabled and an error message should be printed into the log
In the successful case, I would print a message saying what kind of memory pool we are hooking to and what kind of thresholds we are observing.